packages/ckeditor5-media-embed/src/mediaembedcommand.js

```diff
@@ -30,12 +30,12 @@ export default class MediaEmbedCommand extends Command {
 		const model = this.editor.model;
 		const selection = model.document.selection;
 		const schema = model.schema;
-		const position = selection.getFirstPosition();
+		const insertPosition = findOptimalInsertionPosition( selection, model );
 		const selectedMedia = getSelectedMediaModelWidget( selection );
 
-		let parent = position.parent;
+		let parent = insertPosition.parent;
 
-		if ( parent != parent.root ) {
+		if ( parent.isEmpty && model.schema.isBlock( parent ) ) {
```

Comment:
```suggestion
		// The model.insertContent() will remove empty parent (unless it is a $root or a limit).
		if ( parent.isEmpty && !model.schema.isLimit( parent ) ) {
```

I thought the `isBlock()` check will be OK as the behavior suggests. But I wasn't certain why. It turns out that the [insertion logic](https://github.com/ckeditor/ckeditor5/blob/0782dc229381ea185a4d67ec449550ba57a9f1c5/packages/ckeditor5-engine/src/model/utils/insertcontent.js#L560-L583) is a bit different - hence the proposal to change.

I've dug up how this behaves - ATM I'm not sure which is better but let's keep it consistent with `model.insertContent()` checks. I thought it checked for `block`s but it will remove any non-limit empty element on insert.

A "limit" is an element that can contain a selection. In most cases, we do not go past the limit elements. An example is a `<tableCell>` where you can act as like inside editor root. For instance, if you use <kbd>ctrl</kbd> + <kbd>a</kbd> the first use will select all table cell contents.  
The `$root` element is a special case for the limit as you can't go with selection past root.